src/main/java/com/rogermiranda1000/portalgun/PortalGun.java

```diff
@@ -155,6 +179,9 @@ private static void teleportEntities() {
                 synchronized (PortalGun.teleportedEntities) {
                     if (PortalGun.teleportedEntities.containsKey(e)) continue;
                 }
+                synchronized (PortalGun.entityTeleportBlacklist) {
```

Comment:
There's no need for the `synchronized`; `entityTeleportBlacklist` should be loaded and unmodified once the task is started.